app/src/main/java/ch/hikemate/app/model/route/HikesViewModel.kt

```diff
@@ -421,8 +422,15 @@ class HikesViewModel(
    *
    * If [_ongoingLoadingOperations] becomes negative, it is clamped at 0. [setLoading] should always
    * be used in pair, one call to set loading to true, and one to set it to false.
+   *
+   * Additionally, updates [_loadingErrorMessageId] with the [errorMessageId] parameter.
+   *
+   * @param value The new value of the loading state, whether loading (true) or not (false).
+   * @param errorMessageId The ID of the string resource to display to the user if the loading
+   *   operation has failed. This parameter is only considered if [value] is false. This value will
+   *   be set even if other loading operations are still ongoing.
    */
-  private suspend fun setLoading(value: Boolean) =
+  private suspend fun setLoading(value: Boolean, errorMessageId: Int?) =
```

Comment:
I had this debate while writing the function. The thing is, if you set `null` as default value and someone forgets to put an error message for some reason, then it will interpret it as a success.

Furthermore, this function is internal to the view model, it's not used outside of it, so I assumed the users of this function will "know" it and understand they have to pass an error message id to make it an error.

In the end, I personally think it might be better to require the user of the function to explicitly say "there was no problem, take `null` as parameter".

Given my thinking process on this one, what's your call? I can change it if you still believe I should.

Reply:
Yes, I realize the documentation is not very clear, I added some to further clarify what this parameter does.